admin.php

```diff
@@ -7,39 +7,148 @@
+     */
+    const TEMPLATE_DIR = __DIR__ . '/tpl';
+
+    public function getMenuSort()
+    {
+        return self::MENU_SORT;
     }
 
     /**
-     * @return bool true if only access for superuser, false is for superusers and moderators
+     * Admins or higher can access this plugin.
+     * It only has page data that is accessible for admins.
+     * You do not need to be a superuser to access this plugin.
      */
-    public function forAdminOnly() {
+    public function forAdminOnly()
+    {
         return true;
     }
 
     /**
-     * Should carry out any processing required by the plugin.
+     * HTML output (gets generated by twig).
      */
-    public function handle() {
+    public function html()
+    {
+        $pagesWithoutTitle = $this->getPagesWithoutTitle();
+        $vars = $this->getVariablesForTemplate($pagesWithoutTitle);
+        $translations = $this->getTranslations();
+
+        $variables = array_merge($vars, $translations);
+
+        $loader = new Twig_Loader_Filesystem(self::TEMPLATE_DIR);
+        $twig = new Twig_Environment($loader);
+
+        echo $twig->render('admin.tpl', $variables);
+    }
+
+    /**
+     * @return array all pages that do not have a title set.
+     */
+    private function getPagesWithoutTitle() {
+        $indexer = new Doku_Indexer();
+        $allPages = $indexer->getPages();
+
+        foreach ($allPages as $page) {
+            if (p_get_metadata($page)['description']) { // Only get pages with content
+                if (!p_get_metadata($page)['title']) { // Check for pages without a title
+                    // Set variables that are used for the template
+                    $pagesWithoutTitle[] = $page;
+                }
+            }
+        }
+
+        return $pagesWithoutTitle;
+    }
+
+    /**
+     * @param $pagesWithoutTitle array of pages without title.
+     * @return array variables for twig template.
+     */
+    private function getVariablesForTemplate($pagesWithoutTitle) {
+
+        // Generate variables based on page data
+        foreach ($pagesWithoutTitle as $key => $page) {
+            $metadata[] = p_get_metadata($page);
+            $url[] = wl($page, '', true);
+
+            $timeModified = new DateTime('@' . $metadata[$key]['last_change']['date']);
+            $timeModified->add(new DateInterval('PT2H'));
+            $modifiedTimeStamp[] = $timeModified->format(self::TIME_FORMAT) . ' ' . self::TIMEZONE; // Last edit
+        }
+
+        // Put variables to array
+        $variables = array(
+            'pagesWithoutTitle' => $pagesWithoutTitle,
+            'metadata' => $metadata,
+            'urls' => $url,
+            'timestamp' => $modifiedTimeStamp,
+            'imageUrl' => self::EDIT_IMAGE_URL,
+            );
+
+        return $variables;
+    }
+
+    private function getTranslations() {
+
+        // Get translations from DokuWiki
+        $translations = array(
+            'menu' => $this->getLang('menu'),
```

Comment:
Use constants instead of magic strings.

Reply:
What's the point of this at all? The respective `lang.php` already builds a `$lang` variable with all translations. Try to pass that to the template and use it there.

Reply:
Add array to template